library/core/src/str/pattern.rs

```diff
@@ -160,6 +160,19 @@ pub trait Pattern: Sized {
             None
         }
     }
+
+    /// Returns the pattern as utf-8 bytes if possible.
+    fn as_utf8_pattern(&self) -> Option<Utf8Pattern<'_>>;
```

Comment:
Why doesn't this return `None` as a default implementation? As it stands, this seems to cause needless breakage to crates that implement this trait. And even aside from the breakage, as a matter of API design, given that this is an optional optimization, it seems like this _should_ return `None` by default.

Reply:
Yes 100%, sorry. Fix: https://github.com/rust-lang/rust/pull/132113